src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -215,6 +225,7 @@ public function normalize(mixed $object, string $format = null, array $context =
 
             if (null !== $attributeValue && !is_scalar($attributeValue)) {
                 $stack[$attribute] = $attributeValue;
+                continue;
```

Comment:
Why this chance? Isn't this a potential BC break?

Reply:
You are right, this probably breaks BC because the order of elements in the serialized string is changed. Reverting this also takes care of the changes in tests of other components.